code/modules/admin/create_mob.dm

```diff
@@ -21,5 +20,6 @@
 	H.facial_hair_color = H.hair_color
 	H.eye_color = random_eye_color()
 	H.dna.blood_type = random_blood_type()
+	H.dna.generate_uni_identity(H)

```

Comment:
Moved the UI generation at the end so the UI matches the new body characteristics.